presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2396,9 +2416,31 @@ private static Domain buildColumnDomain(ColumnHandle column, List<HivePartition>
         validatePartitionColumns(tableMetadata);
```

Comment:
I tried to find the callers of getNewTableLayout and it led to PrismMetadata, but from there it led nowhere. How is getNewTableLayout used?